cumulusci/core/tests/test_config.py

```diff
@@ -310,519 +252,46 @@ def test_process_github_dependency(self):
-            stderr=io.BytesIO(b''),
-            returncode=0,
-        )
-
-        config = ScratchOrgConfig({'username': 'test', 'created': True}, 'test')
-        config.delete_org()
-
-        self.assertFalse(config.config['created'])
-        self.assertIs(config.config['username'], None)
-
-    def test_delete_org_not_created(self, Command):
-        config = ScratchOrgConfig({'created': False}, 'test')
-        config.delete_org()
-        Command.assert_not_called()
-
-    def test_delete_org_error(self, Command):
-        Command.return_value = mock.Mock(
-            stdout=io.BytesIO(b'An error occurred deleting this org'),
-            stderr=io.BytesIO(b''),
-            returncode=1,
-        )
-
-        config = ScratchOrgConfig({'username': 'test', 'created': True}, 'test')
-        with self.assertRaises(ScratchOrgException):
-            config.delete_org()
-
-    def test_force_refresh_oauth_token(self, Command):
-        Command.return_value = p = mock.Mock(
-            stdout=io.BytesIO(b''),
-            stderr=io.BytesIO(b''),
-            returncode=0,
-        )
-
-        config = ScratchOrgConfig({'username': 'test'}, 'test')
-        config.force_refresh_oauth_token()
-
-        p.run.assert_called_once()
-
-    def test_force_refresh_oauth_token_error(self, Command):
-        Command.return_value = p = mock.Mock(
-            stdout=io.BytesIO(b'error'),
-            stderr=io.BytesIO(b''),
-            returncode=1,
-        )
-
-        config = ScratchOrgConfig({'username': 'test'}, 'test')
-        with self.assertRaises(ScratchOrgException):
-            config.force_refresh_oauth_token()
-
-    def test_refresh_oauth_token(self, Command):
-        result = b'''{
-    "result": {
-        "instanceUrl": "url",
-        "accessToken": "access!token",
-        "username": "username",
-        "password": "password"
-    }
-}'''
-        Command.return_value = mock.Mock(
-            stdout=io.BytesIO(result),
-            stderr=io.BytesIO(b''),
-            returncode=0
-        )
-
-        config = ScratchOrgConfig({'username': 'test'}, 'test')
-        config._scratch_info = {}
-        config._scratch_info_date = datetime.now() - timedelta(days=1)
-        config.force_refresh_oauth_token = mock.Mock()
-
-        config.refresh_oauth_token(keychain=None)
-
-        config.force_refresh_oauth_token.assert_called_once()
-        self.assertTrue(config._scratch_info)
+    def test_list_tasks(self):
+        tasks = self.task_flow_config.list_tasks()
+        self.assertEqual(len(tasks), 3)
+        deploy = [task for task in tasks if task['name'] == 'deploy'][0]
+        self.assertEqual(deploy['description'], 'Deploy Task')
+
+    def test_get_task(self):
+        task = self.task_flow_config.get_task('deploy')
+        self.assertIsInstance(task, BaseConfig)
+        self.assertDictContainsSubset({'description': 'Deploy Task'}, task.config)
+
+    @nose.tools.raises(TaskNotFoundError)
```

Comment:
Let's avoid using utils imported from nose since we have some interest in switching to pytest. For this I'd do:

    with self.assertRaises(TaskNotFoundError):
        self.task_flow_config.get_task('robotic_superstar')